inc/mailcollector.class.php

```diff
@@ -952,11 +957,11 @@ function buildTicket($uid, \Zend\Mail\Storage\Message $message, $options = []) {
 
       // Add to and cc as additional observer if user found
       $ccs = $headers['ccs'];
-      if (is_array($ccs) && count($ccs)) {
+      if (is_array($ccs) && count($ccs) && $this->getField("add_cc_to_observer")) {
          foreach ($ccs as $cc) {
             if ($cc != $requester
                && !Toolbox::inArrayCaseCompare($cc, $blacklisted_emails) // not blacklisted emails
-               && ($tmp = User::getOrImportByEmail($cc) > 0)) {
+               && (($tmp = User::getOrImportByEmail($cc)) > 0)) {
```

Comment:
```suggestion
               && ($tmp = User::getOrImportByEmail($cc)) > 0) {
```

Reply:
revert to previous check version

Reply:
this ```&& ($tmp = User::getOrImportByEmail($to) > 0)``` affected ```true``` value to ```$tmp``` variable while 

 ```&& (($tmp = User::getOrImportByEmail($to)) > 0))``` affected users_id value to ```$tmp``` variable